File: src/main/java/dev/taboo/taboo/commands/Invite.java
package dev.taboo.taboo.commands;

import com.jagrosh.jdautilities.command.CommandEvent;
import com.jagrosh.jdautilities.command.SlashCommand;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.MessageEmbed;
import net.dv8tion.jda.api.entities.User;
import net.dv8tion.jda.api.events.interaction.commands.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.InteractionHook;

import java.time.Instant;

public class Invite extends SlashCommand {

    public Invite() {
        this.name = "invite";
        this.aliases = new String[] { "inv", "invite" };
        this.help = "Invite the bot to your server!";
        this.guildOnly = true;
    }

    @Override
    protected void execute(SlashCommandEvent event) {
        User user = event.getUser();
        InteractionHook hook = event.getHook();
        event.deferReply(true).queue();
        hook.sendMessageEmbeds(inviteEmbed(user)).mentionRepliedUser(false).setEphemeral(false).queue();
    }

    @Override
    protected void execute(CommandEvent event) {
        User author = event.getAuthor();
        Message message = event.getMessage();
        message.replyEmbeds(inviteEmbed(author)).mentionRepliedUser(false).queue();
    }

    private MessageEmbed inviteEmbed(User user) {
        return new EmbedBuilder()
                .setTitle("Invite the bot to your server!")
                .setDescription(
                        "[Click here to invite the bot to your server!]" +
                        "(https://discord.com/api/oauth2/authorize?client_id=892077333878566962&permissions=8&scope=bot%20applications.commands)"
                ).setColor(0x9F90CF)
                .setFooter("Requested by " + user.getAsTag(), user.getEffectiveAvatarUrl())
                .setTimestamp(Instant.now())
                .build();
    }

}

File: src/main/kotlin/dev/taboo/taboo/commands/Invite.kt
package dev.taboo.taboo.commands

import com.jagrosh.jdautilities.command.CommandEvent
import com.jagrosh.jdautilities.command.SlashCommand
import dev.minn.jda.ktx.Embed
import net.dv8tion.jda.api.entities.MessageEmbed
import net.dv8tion.jda.api.entities.User
import net.dv8tion.jda.api.events.interaction.commands.SlashCommandEvent
import java.time.Instant

class Invite: SlashCommand() {

    init {
        name = "invite"
        aliases = arrayOf("inv")
        help = "Invite Taboo to your server"
        guildOnly = true
    }

    override fun execute(event: SlashCommandEvent) {
        val user = event.user
        val hook = event.hook
        event.deferReply(true).queue()
        hook.sendMessageEmbeds(inviteEmbed(user)).mentionRepliedUser(false).queue()
    }

    override fun execute(event: CommandEvent) {
        val author = event.author
        val message = event.message
        message.replyEmbeds(inviteEmbed(author)).mentionRepliedUser(false).queue()
    }

    private fun inviteEmbed(user: User): MessageEmbed {
        val link = "https://discord.com/api/oauth2/authorize?client_id=892077333878566962&permissions=8&scope=bot%20applications.commands"
        return Embed {
            title = "Invite me to your server!"
            description = "[Click here to invite me to your server!]($link)"
            color = 0x9F90CF
            footer {
                name = "Requested by ${user.asTag}"
                iconUrl = user.effectiveAvatarUrl
            }
            timestamp = Instant.now()
        }
    }
}